cms/djangoapps/modulestore_migrator/api.py

```diff
@@ -83,3 +89,32 @@ def get_migration_info(source_keys: list[CourseKey | LibraryLocator]) -> dict:
             named=True,
         )
     }
+
+
+def get_target_block_usage_keys(source_key: CourseKey | LibraryLocator) -> dict[UsageKey, LibraryUsageLocatorV2 | None]:
+    """
+    For given source_key, get a map of legacy block key and its new location in migrated v2 library.
+    """
+    query_set = ModulestoreBlockMigration.objects.filter(overall_migration__source__key=source_key).values_list(
+        'source__key', 'target__key', 'target__learning_package__key'
+    )
+
+    def construct_usage_key(lib_key_str: str, usage_key_str: str) -> LibraryUsageLocatorV2 | None:
+        try:
+            lib_key = LibraryLocatorV2.from_string(lib_key_str)
+        except InvalidKeyError:
+            return None
+        try:
+            # Example: xblock.v1:problem:e9eef38f5f4c49de943c83a2d5170211
+            _, block_type, usage_id = usage_key_str.split(':')
```

Comment:
```suggestion
            _, block_type, block_id = usage_key_str.split(':')
```

`usage_id` is a synonym for `usage_key`. What this slug is is correctly called a `block_id` or, in OLX terms, an `url_name`.